pkg/virtual/workspaces/registry/rest.go

```diff
@@ -549,3 +570,124 @@ func (s *REST) Delete(ctx context.Context, name string, deleteValidation rest.Va
+		if conditions.IsTrue(workspace, tenancyv1alpha1.WorkspaceURLValid) {
+			ctx := genericapirequest.WithCluster(context.TODO(), genericapirequest.Cluster{Name: workspace.ClusterName})
+			shard, err := s.workspaceShardClient.Get(ctx, workspace.Status.Location.Current, metav1.GetOptions{})
+			if err != nil {
+				return nil, wrapError(err)
+			}
+			secret, err := s.coreClient.Secrets(shard.Spec.Credentials.Namespace).Get(ctx, shard.Spec.Credentials.Name, metav1.GetOptions{})
+			if err != nil {
+				return nil, wrapError(err)
+			}
+			data, ok := secret.Data[tenancyv1alpha1.WorkspaceShardCredentialsKey]
+			if !ok {
+				return nil, wrapError(fmt.Errorf("Key '%s' not found in workspace shard Kubeconfig secret", tenancyv1alpha1.WorkspaceShardCredentialsKey))
+			}
+			shardKubeConfig, err := clientcmd.Load(data)
+			if err != nil {
+				return nil, wrapError(fmt.Errorf("Workspace shard Kubeconfig is invalid: %w", err))
+			}
+
+			currentContext := shardKubeConfig.Contexts[shardKubeConfig.CurrentContext]
+			if currentContext == nil {
+				return nil, wrapError(errors.New("Workspace shard Kubeconfig has no current context"))
+			}
+			currentCluster := shardKubeConfig.Clusters[currentContext.Cluster]
+			if currentCluster == nil {
+				return nil, wrapError(fmt.Errorf("Workspace shard Kubeconfig has no cluster corresponding to the current context cluster key: %s", currentContext.Cluster))
+			}
+			currentCluster.Server = workspace.Status.BaseURL
+
+			workspaceContextName := scope + "/" + workspace.Name
+			workspaceConfig := &api.Config{
+				APIVersion:     "v1",
+				Clusters:       map[string]*api.Cluster{workspaceContextName: currentCluster},
+				Contexts:       map[string]*api.Context{workspaceContextName: {Cluster: workspaceContextName}},
+				CurrentContext: workspaceContextName,
+			}
+			dataToReturn, err := clientcmd.Write(*workspaceConfig)
+			if err != nil {
+				return nil, wrapError(err)
+			}
+			return KubeConfig(string(dataToReturn)), nil
+		} else {
+			return nil, wrapError(errors.New("Workspace URL is not valid"))
+		}
+	}
+	return nil, kerrors.NewNotFound(tenancyv1alpha1.SchemeGroupVersion.WithResource("workspaces/kubeconfig").GroupResource(), name)
+}
+
+func (s *KubeconfigSubresourceREST) NamespaceScoped() bool {
+	return false
+}
+
+// New creates a new Workspace log options object
+func (r *KubeconfigSubresourceREST) New() runtime.Object {
+	// TODO - return a resource that represents a log
+	return &tenancyv1alpha1.Workspace{}
+}
+
+// ProducesMIMETypes returns a list of the MIME types the specified HTTP verb (GET, POST, DELETE,
+// PATCH) can respond with.
+func (r *KubeconfigSubresourceREST) ProducesMIMETypes(verb string) []string {
+	// Since the default list does not include "plain/text", we need to
+	// explicitly override ProducesMIMETypes, so that it gets added to
+	// the "produces" section for workspaces/{name}/kubeconfig
+	return []string{
+		"text/plain",
```

Comment:
yaml?

Reply:
Fixed in commit https://github.com/kcp-dev/kcp/pull/375/commits/2c97666d51d66ccc901f7d6f92992f5ba1412a5d

Reply:
Fixed in commit 2c97666d51d66ccc901f7d6f92992f5ba1412a5d